regulations/tasks.py

```diff
@@ -38,15 +33,42 @@ def submit_comment(comment):
 
 
 @contextlib.contextmanager
-def TemporaryDirectory(*args, **kwargs):
+def AttachmentSet(comment):
```

Comment:
I like this -- the code in this function is still a bit hard to understand, but it's nicely hidden away.

I think `TemporaryDirectory` was used as it's the name of a construct in Python 3; if we're switching from that, any reason to keep Cap-camel case?